src/client/containers/dashboard-page/dashboard-page.component.js

```diff
@@ -1,123 +1,246 @@
-              productionStartDate={productionStartDate}
-              productionEndDate={productionEndDate}
-            />
-          </Card>
-          <Card title="Temperature">
-            <LineChartForDashboard
-              tempData={placeholder}
-              strokeGrid={placeholder}
-              strokeAxis={placeholder}
-              minColor={placeholder}
-              maxColor={placeholder}
-              optimalValue={placeholder}
-              strokeWidthRef={placeholder}
-              strokeLine={placeholder}
-              strokeWidthLine={placeholder}
-              ReferanceAreaColor={placeholder}
-            />
-          </Card>
-          <Card title="Humidity">
-            <LineChartForDashboard
-              tempData={placeholder}
-              strokeGrid={placeholder}
-              strokeAxis={placeholder}
-              minColor={placeholder}
-              maxColor={placeholder}
-              optimalValue={placeholder}
-              strokeWidthRef={placeholder}
-              strokeLine={placeholder}
-              strokeWidthLine={placeholder}
-              ReferanceAreaColor={placeholder}
-            />
-          </Card>
-          <Card title="Water PH">
-            <LineChartForDashboard
-              tempData={placeholder}
-              strokeGrid={placeholder}
-              strokeAxis={placeholder}
-              minColor={placeholder}
-              maxColor={placeholder}
-              optimalValue={placeholder}
-              strokeWidthRef={placeholder}
-              strokeLine={placeholder}
-              strokeWidthLine={placeholder}
-              ReferanceAreaColor={placeholder}
-            />
-          </Card>
-          <Card title="Water EC">
-            <LineChartForDashboard
-              tempData={placeholder}
-              strokeGrid={placeholder}
-              strokeAxis={placeholder}
-              minColor={placeholder}
-              maxColor={placeholder}
-              optimalValue={placeholder}
-              strokeWidthRef={placeholder}
-              strokeLine={placeholder}
-              strokeWidthLine={placeholder}
-              ReferanceAreaColor={placeholder}
-            />
-          </Card>
-          <Card title="Water Level">
-            <LineChartForDashboard
-              tempData={placeholder}
-              strokeGrid={placeholder}
-              strokeAxis={placeholder}
-              minColor={placeholder}
-              maxColor={placeholder}
-              optimalValue={placeholder}
-              strokeWidthRef={placeholder}
-              strokeLine={placeholder}
-              strokeWidthLine={placeholder}
-              ReferanceAreaColor={placeholder}
-            />
-          </Card> */}
-        </div>
+    <div className="dashboard">
+      <SidebarMenu text="Dashboard" isActive={false} />
+      <div className="content">
+        <header>
+          <h1>Batch overview</h1>
+        </header>
+        <main>
+          <ProgressBar
```

Comment:
This has to be decomposed. First of all it will become difficult to maintain when the function is so big. Secondly, its not descriptive. A high level component should start out by simply describing to the reader what it contains. It should take about 3-5 seconds to read.

Reply:
I suggest putting all the arrays in variables. Maybe under a common object to hold all the data for the dashboard

Reply:
```
strokeGrid="#fff"
strokeAxis="#808080"
minColor="#FF0000"
maxColor="#FF0000"
optimalValue="#008000"
strokeWidthRef={2}
strokeLine="#000"
strokeWidthLine={2}
ReferanceAreaColor="#808080"
```
These properties should go in an object also. A good rule of thumb is to limit the amount of parameters that are moved around. One might easy forget to insert or add one in the function. This also makes it easier to maintain

Reply:
So as we discussed in class, the main reason that your component is so big is that you have dummy data for the component inlined with the component itself. But since you're waiting for an endpoint to get real data that is going to be refactored anyway, so don't worry about pulling that out in separate variables.

About the changes variables Niels is referencing above, which are essentially about theming your Charts, you should have them in one central place. Normally I would put them in the theming file, but since you need the variables in js and we've not set up an easy way to read the theming file in js I would just put them together in a const containing an object in this file an reference it from there so you're not repeating yourself.

About what we discussed about having more "strict" separation of concerns between "container" and "stateless" components, I was arguing that you could add another layer do you would have a hierarchy like this:

- DashboardPageContainer ("stateful" container component)
  - DashboardPage (stateless functional component)
    - Progressbar (stateless functional component)
    - Card (stateless functional component)
    - LinechartForDashboard (stateless functional component)
    - etc.

**DashboardPageContainer**
Responsible for getting (and possibly combining data from the API and for managing state). No markup, the render function will only render the `<DashboardPage>` component. Should not be attempted mocked in storybook.

**DashboardPage**
Displays a full dashboard page. Essentially the render function will look like what you have in this component at the moment, but you will have to extract all state and put it in the container component, making this a stateless component. The responsibility of this component is to compose together other stateless components into a page layout. Should have a story in storybook.

**Any dependent child components**
This could be card components, charts or any other UI component you need to display. Should all be stateless components, although they can have some local state that is not shared with other component (i.e. disable a form field when a certain radio button is toggled or so on). They should not have side-effects, meaning that they should under no circumstances be calling the API directly. They should communicate what they need up to the container that will supply the data from the API. Should have stories in storybook.

The benefit of this approach is that you get very clear separation of concerns. Your "stateless" components will be pure functions (i.e. no side-effects) which makes them very easy to test, but more essential for us is that it makes them very easy to mock in Storybook. In your particular case you were essentially building a "story" inside the application by adding all of that mock data. If you turn your component into a stateless component you can move all of that mock data into Storybook. The benefit of doing this is that someone who needs to review the DashboardPage in it's entirety (maybe a UX designer) can go find it in Storybook without needing to have knowledge of how to get the application in a state to show that page correctly (i.e. create a user with the right priviliges, seed the right data to database).

Most projects I have worked on don't take this "pure" approach and separate the containers entirely from the markup, but personally I think it's worth-wile.